src/components/map/ha-map.ts

```diff
@@ -155,8 +155,8 @@ export class HaMap extends ReactiveElement {
   }
 
   private _updateMapStyle(): void {
-    const darkMode = this.darkMode ?? this.hass.themes.darkMode ?? false;
-    const forcedDark = this.darkMode ?? false;
+    const darkMode = this.darkMode || this.hass.themes.darkMode || false;
+    const forcedDark = this.darkMode || false;
     const map = this.shadowRoot!.getElementById("map");
```

Comment:
```suggestion
    const darkMode = this.darkMode || this.hass.themes.darkMode ?? false;
    const forcedDark = this.darkMode;
    const map = this.renderRoot.querySelector("#map");
```

Reply:
```
 const forcedDark = this.darkMode;
```

I think this is problematic as some maps (like hui-map-card) still pass in `undefined` to darkMode, which cause `forcedDark` to be undefined, which doesn't seem to behave correctly, I don't think classList.toggle works correctly with `undefined` as the test.

Reply:
Is it not better to be safe and convert it to false?

Reply:
This is the problem with ignoring warnings for `no-incompatible-type-binding`.. 😮‍💨 

Just make the change in `hui-map-card` from `.darkMode=...` to `?darkMode=...` (or wrap the expression in `Boolean()`).  That looks like the only problem location.